core/src/main/scala/org/apache/spark/scheduler/ExecutorDecommissionInfo.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+/**
+ * Provides more detail when an executor is being decommissioned.
+ * @param message Human readable reason for why the decommissioning is happening.
+ * @param isHostDecommissioned Whether the host (aka the `node` or `worker` in other places) is
+ *                             being decommissioned too. Used to infer if the shuffle data might
+ *                             be lost if external shuffle service is enabled.
+ */
+private[spark]
+case class ExecutorDecommissionInfo(message: String, isHostDecommissioned: Boolean) {
+  // Shuffle files are lost when an extecutor is decommissioned if:
+  // - No external shuffle service is present and executor is serving the shuffle data
+  // - It's known that the node (and thus the resident external shuffle service) is also lost
+  def isShuffleLost(externalShuffleServiceEnabled: Boolean): Boolean =
+    !externalShuffleServiceEnabled || isHostDecommissioned
```

Comment:
I think we want to add the check here for the local shuffle block serving as well (once that's integrated). Maybe file a follow up JIRA that depends on both of those JIRAs for tracking.

Reply:
Same question as above: By local shuffle block serving, did you mean https://github.com/apache/spark/pull/28911 ? (And the same line of reasoning for adding this check as well ?)

Reply:
Yes that is exactly what I meant :)